sotodlib/toast/ops/corotator.py

```diff
@@ -86,35 +98,46 @@ def _exec(self, data, detectors=None, **kwargs):
             if obs.telescope.name != "LAT":
                 continue
 
+            el_rad = obs.shared[self.elevation]
+
+            if self.corotator_angle not in obs.shared:
```

Comment:
The `ensure()` method is for detector data (https://github.com/hpc4cmb/toast/blob/ee47b337eccf57f5911893e1b1b7021219ff5fe0/src/toast/observation_data.py#L608) where it can re-use the existing detector data buffer by re-assigning it to different detector sets.